spec/usage/components/transition_spec.rb

```diff
@@ -113,6 +117,44 @@ def page2
     expect(first_content_on_page_1).not_to eq(refreshed_content_on_page_1)
   end
 
+  it "Example 2 - Perform transition from one page to another without page reload when using page history buttons" do
+
+    visit "/my_example_app/page1"
+
+    expect(page).to have_content("My Example App Layout")
+    expect(page).to have_button("Page 1")
+    expect(page).to have_button("Page 2")
+
+    expect(page).to have_content("This is Page 1")
+    expect(page).not_to have_content("This is Page 2")
+
+    element = page.find("#my-div-on-page-1")
+    first_content_on_page_1 = element.text
+
+    page.evaluate_script('document.body.classList.add("not-reloaded")')
+    expect(page).to have_selector("body.not-reloaded")
+
+    click_button("Page 2")
+
+    expect(page).to have_content("My Example App Layout")
+    expect(page).not_to have_content("This is Page 1")
+    expect(page).to have_content("This is Page 2")
+    expect(page).to have_selector("body.not-reloaded")
+
+    # page.evaluate_script('window.history.back()')
+    page.go_back
+
+    element = page.find("#my-div-on-page-1")
+    refreshed_content_on_page_1 = element.text
```

Comment:
This is prone to a race condition again. if `page.go_back` needs longer than finding the div then it might end up with the old content. You can fix this with `expect(page).to have_no_content(first_content_on_page_1)` (`expect(page).not_to have_content...` also works due to dark rspec magic) instead of   `expect(first_content_on_page_1).not_to eq(refreshed_content_on_page_1)`

Reply:
good call!